src/decoder/lattice-incremental-online-decoder.cc

```diff
@@ -98,7 +98,7 @@ typename LatticeIncrementalOnlineDecoderTpl<FST>::BestPathIterator LatticeIncrem
     // it a fatal error for now.
     KALDI_WARN << "No final token found.";
   }
-  if (final_cost_out == NULL)
+  if (final_cost_out != NULL)
     *final_cost_out = best_final_cost;
   return BestPathIterator(best_tok, this->NumFramesDecoded() - 1);
 }
```

Comment:
yes but please do the same for lattice-incremental-decoder.cc

Reply:
I do not see this error in lattice-incremental-decoder.cc, where exactly?

Reply:
oh sorry, it's just here.